Add a ScrollBar to the skilllist

Ewwweee opened the pull request:
feat(EventualSkills) : add a scrollbar to the list of skills in the menu 

The scrollbar appears only if the height of the skills exceed the height of the window

It was made to resolve the issue #1 .

Ewwweee committed:
Update EventualSkillsTraining.ui

feat(skillList) : add a ScrollableArea to the list of the skills

The scrollBar is invisible until there are enough skill in the skill Menu (when the list of the buttons take too much place, a scrollbar appears)

It was made to resolve the issue #1

Ewwweee committed:
Merge pull request #1 from Ewwweee/Ewwweee-patch-1

Update EventualSkillsTraining.ui

Ihebsmai committed:
ihebtrying

jdrueckert commented:
@Ewwweee it seems like someone committed to your fork. Is "Ihebsmai" someone you're familiar with?
In any case the file they added needs to be removed again before we can consider merging this.

Ewwweee committed:
msg

Ewwweee commented:
Hello, I modified my pull request (my friend commit a change in the wrong repo). Do I need to open a new pull request ?

jdrueckert commented:
Hi @Ewwweee,
I just tested your PR in-game and that's what the result looks like:
![image](https://github.com/Terasology/EventualSkills/assets/29981695/dbfc70e7-061f-4eee-9959-bfd8128e73fb)

Most problematic here is that when I reduce my window size, I cannot even see the skill buttons anymore:
![image](https://github.com/Terasology/EventualSkills/assets/29981695/0f0b6f31-2326-4089-94f7-e033d2378761)

IIRC what you shared on Discord looked different (and way better) - not sure what happened/changed? :thinking: 
Anyway, in this state I don't think we can merge this so please revisit your code changes.

Ewwweee committed:
Scrollbar V2

Ewwweee committed:
Merge pull request #2 from Ewwweee/issue2

Scrollbar V2

Ewwweee commented:
Hello ! I fixed the pull request so normally it should work this time  (never 2 without 3) ! . I added some skills to test the scrollbar in EventualSkills\prefab. It should look like this image if I correctly did my pull request. 
![image](https://github.com/Terasology/EventualSkills/assets/98258435/a17347a3-1f8c-469d-ad1c-8872053a403a)

jdrueckert commented:
Nice! Now it works perfectly fine :+1: 
Please remove the dummy skills you added for testing again and then we can get this merged :smiley:

Ewwweee committed:
remove dumb skills

Ewwweee committed:
Merge pull request #3 from Ewwweee/issue2

remove dumb skills

Ewwweee commented:
Hello, thanks for your feedback. I removed the dummy skills so normally it should be ok.

jdrueckert commented:
Sorry for the delay, between being sick and the holidays, time just flew by :see_no_evil: 

I just gave this another test and noticed the following when changing my window size:
![image](https://github.com/Terasology/EventualSkills/assets/29981695/b5951aad-47b0-4d71-8ccd-8b5cbb0ee136)
As per your changes the skill learning screen has fix dimensions (height and width), causing it to not reduce its height (or width) if the window is smaller.
However, this issue is independent of the screen size being oriented at the number of skills which was resolved using the scrollbar, so I'd argue we can track this in a follow-up issue. 

I still need to request two changes before merging, though:
1. please adjust your PR title, as per our [contribution guidelines](https://github.com/MovingBlocks/Terasology/blob/develop/.github/CONTRIBUTING.md), we try to follow conventional commits
  (please note that this is not necessary for every commit of your PR as we [squash-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) it, because that results in a single commit on the module's `develop` branch based on your PR's title)
2. please remove the `logback-test.xml` file, we don't need that checked-in to avoid the module behaving differently than the others. if we want to change the current default, this should be done generally for all modules.
